### ISSUE TITLE
arcas file is empty?

### ISSUE THREAD

**Issue by carlosonunez-vmw** [vmware-tanzu/service-installer-for-vmware-tanzu#26]
**Reference Architecture**: TKO on vSphere backed by VDS

## What Happened?

I tried to run the command described in the README:

```
arcas --env vsphere --file /tmp/arcas.json --avi_configuration --tkg_mgmt_configuration --shared_service_configuration --workload_preconfig --workload_deploy --deploy_extentions
```

but it does nothing because the `arcas` tool that the command above is using is an empty file:

```
root@photon-machine [ ~ ]# file $(which arcas)
/usr/bin/arcas: empty
```

## What Was Expected?

For something to happen

## How to Reproduce

1. Create an Arcas appliance from the OVA
2. SSH in
3. Run `file $(which arcas)`

## Versions

```
Version: 1.2.0.50
Build: 19794220
```

**Comment by tasmiya-bano** [comment]
@carlosonunez-vmw Could you please confirm if you generated the json file for your environment using the steps mentioned in the document - https://github.com/vmware-tanzu/service-installer-for-vmware-tanzu/blob/main/docs/product/release/vSphere%20-%20Backed%20by%20VDS/TKGm/TKOonVsphereVDStkg.md (Step 4 in "Deploy Tanzu for Kubernetes Operations" section) ?

SIVT User Interface saves the json file to "/opt/vmware/arcas/src/vsphere-dvs-tkgm.json"  location [OR] if you have created this file manually, you need to copy it SIVT and provide the path in above command you have mentioned.

Hope this helped, let me know if you have further queries.

**Comment by carlosonunez-vmw** [comment]
i created the file using the template on the page locally inside of the appliance.

-- Carlos

On May 19, 2022, at 03:50, tasmiya-bano ***@***.***> wrote:

﻿

@carlosonunez-vmw<https://github.com/carlosonunez-vmw> Could you please confirm if you generated the json file for your environment using the steps mentioned in the document - https://github.com/vmware-tanzu/service-installer-for-vmware-tanzu/blob/main/docs/product/release/vSphere%20-%20Backed%20by%20VDS/TKGm/TKOonVsphereVDStkg.md (Step 4 in "Deploy Tanzu for Kubernetes Operations" section) ?

SIVT User Interface saves the json file to "/opt/vmware/arcas/src/vsphere-dvs-tkgm.json" location [OR] if you have created this file manually, you need to copy it SIVT and provide the path in above command you have mentioned.

Hope this helped, let me know if you have further queries.

**Comment by tasmiya-bano** [comment]
Could you please cross check the name of the file you created ?

Looks like it was not saved with "arcas.json" file name.

**Comment by carlosonunez-vmw** [comment]
But the issue is that the "arcas" binary itself is empty.

-- Carlos

On May 19, 2022, at 13:15, tasmiya-bano ***@***.***> wrote:

﻿

Could you please cross check the name of the file you created ?

Looks like it was not saved with "arcas.json" file name.

**Comment by tasmiya-bano** [comment]
yes, SIVT OVA doesn't have arcas binary. arcas is running is service in SIVT OVA and we have followed CLI based automation. 

You can refer /opt/vmware/arcas/src/cli.py file for more details.

**Comment by carlosonunez-vmw** [comment]
That's confusing. Step 4 of the instructions states to use arca. Is this out of date?

-- Carlos

**Comment by rashikwal** [comment]
@carlosonunez-vmw "arcas" is a service running inside the SIVT OVA. You can type "arcas --help" command to get more idea on the available flags.
For starters, "arcas --env <vsphere/vcf/vmc> --file <complete-path-to-the-input-json-file>" are the mandatory flags to be provided by the user. We also have a UI that can help in generation of the input file.

To check the status of arcas service, you can run "systemctl status arcas.service" on your SIVT ova.
We can connect offline if you need more clarity on the design of the product.

**Comment by carlosonunez-vmw** [comment]
I was able to work around this by doing the below from inside of the SIVT:

```sh
pip install /opt/vmware/arcas
```

**Comment by sunshetty** [comment]
This issue is addressed in SIVT 2.0 release. Closing the issue.
